pytorch_lightning/strategies/ddp.py

```diff
@@ -152,6 +154,17 @@ def setup(self, trainer: "pl.Trainer") -> None:
         if trainer_fn == TrainerFn.FITTING:
             self.configure_ddp()
 
+        # set up optimizers after the wrapped module has been moved to the device
+        self.setup_optimizers(trainer)
+        self.setup_precision_plugin()
+        optimizers_to_device(self.optimizers, self.root_device)
```

Comment:
`setup_optimizers` returns early if the trainer isn't fitting or tuning.
but `setup_precision_plugin` looks up the model, optimizer, and lr scheduler configs.

are all precision plugins built to handle optimizers/LR schedulers not being available outside of fitting/tuning? 

https://github.com/PyTorchLightning/pytorch-lightning/blob/b5fe056765ce71b237c641a1b159c346e5e8bd33/pytorch_lightning/strategies/strategy.py#L146-L152